### PR TITLE
Introduces history manager screen for medical manager

### PR DESCRIPTION
**Pull request by mtabishk**
https://user-images.githubusercontent.com/61789893/193395332-ccc84651-144f-4287-b319-02c291c11246.mp4

1. Added History Manager UI
2. API Integration with backend

### CODE REVIEW

File: api/medical/models.py [hunk 1]
```diff
@@ -55,9 +55,10 @@ class Appointment(models.Model):
         settings.AUTH_USER_MODEL, on_delete=models.CASCADE)
     date = models.DateField(null=True)
     time = models.TimeField(null=True)
-    start_time = models.TimeField(null = True)
-    end_time = models.TimeField(null = True)
-    status = models.CharField(choices=APPOINTMENT_STATUS, default="IN PROGRESS", max_length=30)
+    start_time = models.TimeField(null=True)
+    end_time = models.TimeField(null=True)
+    status = models.CharField(
+        choices=APPOINTMENT_STATUS, default="IN PROGRESS", max_length=30)
```
Comment (ashutoshc8101) [inline api/medical/models.py:61]: Changes not required

File: api/medical/models.py [hunk 2]
```diff
@@ -69,8 +70,6 @@ class MedicalHistory(models.Model):
         settings.AUTH_USER_MODEL, on_delete=models.CASCADE)
     doctor = models.ForeignKey(Doctor, on_delete=models.CASCADE)
     date = models.DateField(null=True)
-    time = models.TimeField(null=True)
```
Comment (ashutoshc8101) [inline api/medical/models.py:72]: Don't remove the time field.

File: api/medical/views.py [hunk 3]
```diff
@@ -233,13 +233,14 @@ def post(self, request):
...
+            name=name,
+            designation=profession,
+            phone=contact
+        )
+
+        staff.save()
+        return Response(status=200, data={'msg': 'Staff added successfully'})
 
-        staff.save();
-        return Response(status = 200, data={ 'msg': 'Staff added successfully' })
```
Comment (ashutoshc8101) [inline api/medical/views.py:242]: Changes not required.

File: api/medical/views.py [hunk 4]
```diff
@@ -306,26 +307,28 @@ def post(self, request):
...
+
 class AppointmentsPending(APIView):
     permission_classes = [IsMedicalManager]
 
     def get(self, request):
         now = datetime.datetime.now()
 
-        appointments = Appointment.objects.filter(status = "ACCEPTED", date__gt=now.date()).all()
+        appointments = Appointment.objects.filter(
+            status="ACCEPTED", date__gt=now.date()).all()
```
Comment (ashutoshc8101) [inline api/medical/views.py:326]: don't push changes that only change the indentation.

File: app/env.symlink [hunk 5]
```diff
@@ -0,0 +1 @@
+../.env
```
Comment (ashutoshc8101) [inline app/env.symlink:1]: Don't push changes to `env.symlink`

File: app/lib/main_dev.dart [hunk 6]
```diff
@@ -21,6 +21,7 @@ Future main() async {
 
   await Firebase.initializeApp(
     options: DefaultFirebaseOptions.currentPlatform,
+    name: 'ira',
```
Comment (ashutoshc8101) [inline app/lib/main_dev.dart:24]: Why is this required?

File: app/lib/screens/medical/manager/history/doctor_model.dart [hunk 7]
```diff
@@ -0,0 +1,38 @@
...
+  final int id;
+  final String name;
+  final String phone;
+  final String specialization;
+  final String mail;
+  final String date;
+  // ignore: non_constant_identifier_names
+  final String start_time;
+  // ignore: non_constant_identifier_names
+  final String end_time;
```
Comment (ashutoshc8101) [inline app/lib/screens/medical/manager/history/doctor_model.dart:22]: Use `camelCase` here.

File: app/lib/screens/medical/manager/history/doctor_model.dart [hunk 8]
```diff
@@ -0,0 +1,38 @@
...
+    this.id,
+    this.name,
+    this.phone,
+    this.specialization,
+    this.mail,
+    this.date,
+    this.start_time,
+    this.end_time,
+    this.details,
+  );
```
Comment (ashutoshc8101) [inline app/lib/screens/medical/manager/history/doctor_model.dart:12]:
```suggestion
  DoctorModel({
    required this.id,
    required this.name,
    required this.phone,
    required this.specialization,
    required this.mail,
    required this.date,
    required this.start_time,
    required this.end_time,
    required this.details,
    }
  );
```

File: app/lib/screens/medical/manager/history/history_manager.dart [hunk 9]
```diff
@@ -0,0 +1,168 @@
...
+        ],
+      ),
+    );
+  }
+}
+
+class StudentModel {
+  // ignore: non_constant_identifier_names
+  final String first_name;
+  // ignore: non_constant_identifier_names
```
Comment (ashutoshc8101) [inline app/lib/screens/medical/manager/history/history_manager.dart:152]: Use camelCase

**Review by ashutoshc8101 — CHANGES_REQUESTED** [review]
Left few comments and dont push `app/flutter_04.png`

**Comment by ashutoshc8101** [comment]
@Jeetu303 Please review video.

**Comment by Jeetu303** [comment]
Seems good
Go ahead

File: app/lib/main_dev.dart (hunk 6 above)
Comment (mtabishk) [reply app/lib/main_dev.dart:24]: It was required on my side. There was some issue "Duplicate Firebase APP"

Comment (ashutoshc8101) [reply app/lib/main_dev.dart]: Remove this field if this doesn't apply to all contributors.

File: app/.env.symlink [hunk 10]
```diff
@@ -1 +0,0 @@
-../.env
```
Comment (ashutoshc8101) [inline app/.env.symlink:1]: This file shouldn't be deleted.

**Review by ashutoshc8101 — CHANGES_REQUESTED** [review]
Not all comments are properly addressed. Also, flutter analyze is still failing. Ask for review, once all comments are properly addressed.

File: app/lib/screens/medical/manager/history/history_manager.dart (hunk 9 above)
Comment (mtabishk) [reply app/lib/screens/medical/manager/history/history_manager.dart]: Done

File: app/lib/screens/medical/manager/history/doctor_model.dart (hunk 7 above)
Comment (mtabishk) [reply app/lib/screens/medical/manager/history/doctor_model.dart]: Done

File: app/env.symlink (hunk 5 above)
Comment (mtabishk) [reply app/env.symlink]: Okay

File: api/medical/views.py (hunk 4 above)
Comment (mtabishk) [reply api/medical/views.py:326]: Done

File: api/medical/views.py (hunk 3 above)
Comment (mtabishk) [reply api/medical/views.py:242]: Done

File: api/medical/models.py (hunk 2 above)
Comment (mtabishk) [reply api/medical/models.py]: Done

File: api/medical/models.py (hunk 1 above)
Comment (mtabishk) [reply api/medical/models.py:61]: Done

File: app/lib/screens/medical/manager/history/doctor_model.dart (hunk 8 above)
Comment (mtabishk) [reply app/lib/screens/medical/manager/history/doctor_model.dart]: Why is this required?

File: app/.env.symlink (hunk 10 above)
Comment (mtabishk) [reply app/.env.symlink:1]: Okay. I have added .env.symlink from ira main branch

File: app/lib/main_dev.dart (hunk 6 above)
Comment (mtabishk) [reply app/lib/main_dev.dart]: Removed

**Comment by mtabishk** [comment]
@ashutoshc8101 Flutter analyze has no issues on my end. I have added .env.symlink from ira main branch and addressed all comments
Screenshot 2022-10-01 133640

File: app/.env.symlink (hunk 10 above)
Comment (ashutoshc8101) [reply app/.env.symlink:1]: This file should be a symlink. Ideally, changes to this file shouldn't be even here in diff.

**Review by ashutoshc8101 — CHANGES_REQUESTED** [review]

**Review by ashutoshc8101 — APPROVED** [review]
